src/vaadin-grid-column.html

```diff
@@ -313,10 +313,6 @@
 
     /** @private */
     __setColumnTemplateOrRenderer(template, renderer, cells) {
-      // no renderer or template needed in a hidden column
-      if (this.hidden) {
-        return;
-      }
```

Comment:
The change seems to affect when the hidden state changes, so this condition was causing header templates not to render. The problem with this condition is that the observer does not listen to hidden changes, so even if the column / group becomes visible this is never called again. Adding hidden to the observer also didn't improve things, as there are some other checks that prevent setting the template.

The condition here was introduced as part of this bugfix: https://github.com/vaadin/vaadin-grid/pull/2146. However it seems the fix was not covering all cases, so a another fix was added later: https://github.com/vaadin/vaadin-grid/pull/2200. It seems that with the second fix the first one is not needed anymore, so it looks like we can remove it. The other concern is doing the extra work of rendering the cells even though they are hidden, though that was already the case before the first fix.